IHP/ModelSupport.hs

```diff
@@ -600,9 +600,15 @@ ids :: (HasField "id" record id) => [record] -> [id]
 ids records = map (getField @"id") records
 {-# INLINE ids #-}
 
+-- | The error message of a validator can be either a plain text value or a HTML formatted value
+data Violation
+    = TextViolation { message :: !Text } -- ^ Plain text validation error, like "cannot be empty"
+    | HtmlViolation { message :: !Text } -- ^ HTML formatted, already pre-escaped validation error, like "Invalid, please <a href="http://example.com">check the documentation</a>"
+    deriving (Eq, Show)
+
 -- | Every IHP database record has a magic @meta@ field which keeps a @MetaBag@ inside. This data structure is used e.g. to keep track of the validation errors that happend.
 data MetaBag = MetaBag
-    { annotations            :: ![(Text, Text)] -- ^ Stores validation failures, as a list of (field name, error) pairs. E.g. @annotations = [ ("name", "cannot be empty") ]@
+    { annotations            :: ![(Text, Violation)] -- ^ Stores validation failures, as a list of (field name, error) pairs. E.g. @annotations = [ ("name", "cannot be empty") ]@
```

Comment:
Not sure, but maybe this should be:

```suggestion
    { annotations            :: ![(Text, Violation)] -- ^ Stores validation failures, as a list of (field name, error) pairs. E.g. @annotations = [ ("name", TextViolation "cannot be empty") ]@
```

Reply:
Good catch, fixed via https://github.com/digitallyinduced/ihp/commit/70738b2541a4331c55ed74f7bfdb979ae990f815 👍